components/hab/src/cli_v4/pkg/demote.rs

```diff
@@ -0,0 +1,62 @@
+// Implementation of `hab pkg demote` command
+
+use clap_v4 as clap;
+
+use clap::Parser;
+
+use habitat_common::{cli::PACKAGE_TARGET_ENVVAR,
+                     ui::UI};
+
+use habitat_core::{package::{target,
+                             PackageIdent,
+                             PackageTarget},
+                   ChannelIdent};
+
+use crate::{cli_v4::utils::{AuthToken,
+                            BldrUrl},
+            command::pkg::demote,
+            error::Result as HabResult};
+
+#[derive(Debug, Clone, Parser)]
+#[command(arg_required_else_help = true,
+          help_template = "{name} {version} {author-section} {about-section} \n{usage-heading} \
+                           {usage}\n\n{all-args}\n")]
+pub(crate) struct PkgDemoteOptions {
+    #[command(flatten)]
+    bldr_url: BldrUrl,
+
+    /// Fully Qualified package identifier for the package
```

Comment:
This was `A fully qualified package identifier (ex: core/busybox-static/1.42.2/20170513215502)`. It's nice to have the example.